artificial-scrum-master/apps/artificial.scrum.master/Artificial.Scrum.Master.ScrumProjectIntegration/Infrastructure/ScrumServiceHttpClient/ProjectHttpClientWrapper.cs

```diff
@@ -0,0 +1,120 @@
+using Artificial.Scrum.Master.ScrumProjectIntegration.Settings;
+using Artificial.Scrum.Master.ScrumProjectIntegration.Utilities;
+
+namespace Artificial.Scrum.Master.ScrumProjectIntegration.Infrastructure.ScrumServiceHttpClient
+{
+    public class ProjectHttpClientWrapper : IProjectHttpClientWrapper
+    {
+        private readonly ScrumManagementServiceSettings _settings;
+
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly IJwtDecoder _jwtDecoder;
+        private readonly IUserTokensRepository _userTokensRepository;
+
+        public ProjectHttpClientWrapper(ScrumManagementServiceSettings settings, IHttpClientFactory httpClientFactory,
+            IUserTokensRepository userTokensRepository, IJwtDecoder jwtDecoder)
+        {
+            _httpClientFactory = httpClientFactory;
+            _userTokensRepository = userTokensRepository;
+            _jwtDecoder = jwtDecoder;
+            _settings = settings;
+        }
+
+        public async Task<WrapperHttpResponse> GetHttpRequest(string userId, string url)
+        {
+            var tokens = await _userTokensRepository.GetUserAccessTokens(userId);
+            if (tokens is null)
+            {
+                throw new ProjectResourceNotFoundException($"Credentials of user:{userId} not found");
+            }
+
+            var isValid = ValidateAccessToken(tokens.Value.AccessToken);
+            if (!isValid)
+            {
+                tokens = await RefreshUserTokens(userId, tokens.Value);
+            }
+
+            var httpClient = _httpClientFactory.CreateClient();
+            httpClient.DefaultRequestHeaders.Add("Authorization", $"Bearer {tokens.Value.AccessToken}");
+
+            var httpResponse = await httpClient.GetAsync(url);
+            var result = await httpResponse.Content.ReadAsStringAsync();
+
+            return new WrapperHttpResponse
+            {
+                Response = result,
+                StatusCode = httpResponse.StatusCode
+            };
+        }
+
+        public async Task<WrapperHttpResponse> PostHttpRequest(string userId, string url,
+            Dictionary<string, string> parameters)
+        {
+            var tokens = await _userTokensRepository.GetUserAccessTokens(userId);
+            if (tokens is null)
+            {
+                throw new ProjectResourceNotFoundException($"Credentials of user:{userId} not found");
+            }
+
+            var isValid = ValidateAccessToken(tokens.Value.AccessToken);
+            if (!isValid)
+            {
+                tokens = await RefreshUserTokens(userId, tokens.Value);
+            }
+
+            var httpClient = _httpClientFactory.CreateClient();
+            httpClient.DefaultRequestHeaders.Add("Authorization", $"Bearer {tokens.Value.AccessToken}");
+
+            var content = new FormUrlEncodedContent(parameters);
+            var httpResponse = await httpClient.PostAsync(url, content);
```

Comment:
czemu przekazujesz formContent zamiast zwykłego obiektu? To api nie przyjmuje zwykłego jsona? Bo ten sposób jest ze średniowiecza (starych formsów)